src/Raven.Client/Documents/Operations/CompareExchange/CompareExchangeValue.cs

```diff
@@ -25,5 +32,6 @@ internal interface ICompareExchangeValue
         public string Key { get; }
         public long Index { get; internal set; }
         public object Value { get; }
+        public IMetadataDictionary Metadata { get; }
```

Comment:
Here, when accessed, create a new instance so users have easier time setting this